src/lerobot/processor/pipeline.py

```diff
@@ -323,107 +353,71 @@ class RobotProcessor(ModelHubMixin):
     to_transition: Callable[[dict[str, Any]], EnvTransition] = field(
         default_factory=lambda: _default_batch_to_transition, repr=False
     )
-    to_output: Callable[[EnvTransition], dict[str, Any] | EnvTransition] = field(
-        default_factory=lambda: _default_transition_to_batch, repr=False
+    to_output: Callable[[EnvTransition], TOutput] = field(
+        # Cast is necessary here: Working around Python type-checker limitation.
+        # _default_transition_to_batch returns dict[str, Any], but we need it to be TOutput
+        # for the generic to work. When no explicit type is given, TOutput defaults to dict[str, Any],
+        # making this cast safe.
+        default_factory=lambda: cast(Callable[[EnvTransition], TOutput], _default_transition_to_batch),
+        repr=False,
     )
 
     # Processor-level hooks for observation/monitoring
     # Hooks do not modify transitions - they are called for logging, debugging, or monitoring purposes
     before_step_hooks: list[Callable[[int, EnvTransition], None]] = field(default_factory=list, repr=False)
     after_step_hooks: list[Callable[[int, EnvTransition], None]] = field(default_factory=list, repr=False)
 
-    def __call__(self, data: EnvTransition | dict[str, Any]):
+    def __call__(self, data: dict[str, Any]) -> TOutput:
```

Comment:
Any reason for the deletion of EnvTrasition?
```suggestion
    def __call__(self, data: EnvTransition) -> TOutput:
```